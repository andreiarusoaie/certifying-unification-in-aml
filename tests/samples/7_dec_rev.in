variables: x, y, z
symbols: f, a, h, b, c
problem: f(h(a, x), b) =? f(z, h(c, y))
