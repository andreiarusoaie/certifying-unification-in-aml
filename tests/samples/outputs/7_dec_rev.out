Found Maude version: 3.0beta1

[1m[95mProof of:[94m (f(z, h(c, y)) or f(h(a, x), b) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13)))  [0m
(1)(âˆƒ v(9) . (f(h(a, x), b) <--> v(9) and (v(9) === f(h(a, x), b)))) [e-subst] ;

(2)((f(h(a, x), b) <--> âˆƒ v(9) . v(9) and (v(9) === f(h(a, x), b))) <--> âˆƒ v(9) . (f(h(a, x), b) <--> v(9) and (v(9) === f(h(a, x), b)))) [e-scope] ;

(3)(f(h(a, x), b) <--> âˆƒ v(9) . v(9) and (v(9) === f(h(a, x), b))) [modus-ponens(1, 2)] ;

(4)(âˆƒ v(9) . (f(z, h(c, y)) <--> v(9) and (v(9) === f(z, h(c, y))))) [e-subst] ;

(5)((f(z, h(c, y)) <--> âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y)))) <--> âˆƒ v(9) . (f(z, h(c, y)) <--> v(9) and (v(9) === f(z, h(c, y))))) [e-scope] ;

(6)(f(z, h(c, y)) <--> âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y)))) [modus-ponens(4, 5)] ;

(7)(f(z, h(c, y)) or f(h(a, x), b) <--> (âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y)))) or (âˆƒ v(9) . v(9) and (v(9) === f(h(a, x), b)))) [eqv-or(3, 6)] ;

(8)((âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y)))) or (âˆƒ v(9) . v(9) and (v(9) === f(h(a, x), b))) <--> âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y))) or v(9) and (v(9) === f(h(a, x), b))) [e-collapse] ;

(9)(v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b))) <--> v(9) and (v(9) === f(z, h(c, y))) or v(9) and (v(9) === f(h(a, x), b))) [and-or-distr] ;

(10)((âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) <--> âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y))) or v(9) and (v(9) === f(h(a, x), b))) [e-context(9)] ;

(11)(f(z, h(c, y)) or f(h(a, x), b) <--> âˆƒ v(9) . v(9) and (v(9) === f(z, h(c, y))) or v(9) and (v(9) === f(h(a, x), b))) [eqv-tranz(7, 8)] ;

(12)(f(z, h(c, y)) or f(h(a, x), b) <--> âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) [eqv-tranz(11, 10)] ;

(13)((v(9) === f(h(a, x), b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [e-gen] ;

(14)((v(9) === f(h(a, x), b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [and-eqv-intro(13)] ;

(15)((âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [e-scope] ;

(16)((v(9) === f(h(a, x), b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [eqv-tranz(14, 15)] ;

(17)((v(9) === f(z, h(c, y))) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) [e-gen] ;

(18)((v(9) === f(z, h(c, y))) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) [and-eqv-intro(17)] ;

(19)((âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) [e-scope] ;

(20)((v(9) === f(z, h(c, y))) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) [eqv-tranz(18, 19)] ;

(21)((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)) <--> (âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) or (âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b))) [eqv-or(16, 20)] ;

(22)((âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y))) or (âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y)) or (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [e-collapse] ;

(23)((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y)) or (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [eqv-tranz(21, 22)] ;

(24)(((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13))) <--> (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y)) or (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [and-or-distr] ;

(25)((âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(12) ; v(13) . (v(9) === f(v(12), v(13))) and (v(12) === z) and (v(13) === h(c, y)) or (v(9) === f(v(12), v(13))) and (v(12) === h(a, x)) and (v(13) === b)) [e-context(24)] ;

(26)((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [eqv-tranz(23, 25)] ;

(27)((âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) <--> âˆƒ v(9) . v(9) and (âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13))))) [e-context(26)] ;

(28)(v(9) and (âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-scope] ;

(29)((âˆƒ v(9) . v(9) and (âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13))))) <--> âˆƒ v(9) . âˆƒ v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-context(28)] ;

(30)((âˆƒ v(9) . âˆƒ v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(9) ; v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-set] ;

(31)((âˆƒ v(9) . v(9) and (âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13))))) <--> âˆƒ v(9) ; v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [eqv-tranz(29, 30)] ;

(32)(((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(9) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-scope] ;

(33)((âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) <--> âˆƒ v(12) ; v(13) . âˆƒ v(9) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-context(32)] ;

(34)((âˆƒ v(12) ; v(13) . âˆƒ v(9) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(9) ; v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [e-set] ;

(35)((âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) <--> âˆƒ v(9) ; v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [eqv-tranz(34, 33)] ;

(36)(âˆƒ v(9) . (f(v(12), v(13)) <--> v(9) and (v(9) === f(v(12), v(13))))) [e-subst] ;

(37)((f(v(12), v(13)) <--> âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13)))) <--> âˆƒ v(9) . (f(v(12), v(13)) <--> v(9) and (v(9) === f(v(12), v(13))))) [e-scope] ;

(38)(f(v(12), v(13)) <--> âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13)))) [modus-ponens(36, 37)] ;

(39)((âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [e-context(38)] ;

(40)((âˆƒ .Vars . âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) <--> âˆƒ .Vars . âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [e-context(39)] ;

(41)((âˆƒ .Vars . âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [e-set] ;

(42)((âˆƒ .Vars . âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) [e-set] ;

(43)((âˆƒ .Vars . âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [eqv-tranz(41, 40)] ;

(44)((âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [eqv-tranz(43, 42)] ;

(45)((âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) <--> âˆƒ v(9) ; v(12) ; v(13) . v(9) and ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (v(9) === f(v(12), v(13)))) [eqv-tranz(27, 31)] ;

(46)((âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and (âˆƒ v(9) . v(9) and (v(9) === f(v(12), v(13))))) [eqv-tranz(45, 35)] ;

(47)((âˆƒ v(9) . v(9) and ((v(9) === f(z, h(c, y))) or (v(9) === f(h(a, x), b)))) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) [eqv-tranz(44, 46)] ;

(48)(f(z, h(c, y)) or f(h(a, x), b) <--> âˆƒ v(12) ; v(13) . ((v(12) === z) and (v(13) === h(c, y)) or (v(12) === h(a, x)) and (v(13) === b)) and f(v(12), v(13))) [eqv-tranz(12, 47)] ;
[94mChecked: [0m [1m[92m true
[0m
